src/modelinstance.cpp

```diff
@@ -505,11 +505,8 @@ Status ModelInstance::reloadModel(const ModelConfig& config, const DynamicModelP
 
 Status ModelInstance::recoverFromReloadingError(const Status& status) {
     if (status == StatusCode::RESHAPE_ERROR) {
```

Comment:
Here we only unload the model if the error is RESHAPE_ERROR. This means we only do it when exception occurs during `network->reshape()` call. What if there will be another kind of error, for example CANNOT_LOAD_INTO_DEVICE (exception during LoadNetwork call)? I think we should try to recover from any kind of error, so I'd suggest to remove this if statement.